scripts/s2n_safety_macros.py

```diff
@@ -722,7 +725,16 @@ def push_doc(args):
             docs += push_doc(args)
             header += push_macro(args)
 
+def cleanup(contents):
+    # Remove any unnecessary generated "X_GUARD_X"s, like "RESULT_GUARD_RESULT"
+    for context in CONTEXTS:
+        x_guard = "{name}_GUARD".format_map(context)
+        x_guard_x = "{name}_GUARD_{name}".format_map(context)
+        contents = contents.replace(x_guard_x, x_guard)
+    return contents
```

Comment:
I'm open to suggestions if you have ideas, but multiple existing parts of the code can produce RESULT_GUARD_RESULTs. [This](https://github.com/aws/s2n-tls/blob/main/scripts/s2n_safety_macros.py#L467-L470) is a macro that produces a RESULT_GUARD_RESULT. It makes sense for all context cases except RESULT. To fix it at the "source", I'd  probably need to special case both [writing the macro](https://github.com/aws/s2n-tls/blob/main/scripts/s2n_safety_macros.py#L639) and [writing the documentation](https://github.com/aws/s2n-tls/blob/main/scripts/s2n_safety_macros.py#L668). Rather than trying to fix the problem everywhere, acknowledging that this is a basic s/X_GUARD_X/X_GUARD/ problem seems more readable / less involved.

Reply:
yeah that's fair. sounds good.

Reply:
I'm not sure if this is nicer, but you could maybe add a guard_result value to the args dictionary [here](https://github.com/aws/s2n-tls/blob/31d99d2226fe352d3989b92378d8595a07711427/scripts/s2n_safety_macros.py#L684) like this:

```
'guard_result': f"{context['prefix']}GUARD{'_RESULT' if context['name'] != 'RESULT' else ''}"
```

Then, you could use {guard_result} instead of {prefix}GUARD_RESULT in the precondition/postcondition macros.

But yeah, I'm not sure if this is any better than the brute force method.